File: API/Biblioteca/Models/Livro.cs
namespace API.Biblioteca.Models;

public class Livro{

    public string Titulo { get; set; }
    public string Autor { get; set; }
    public string Editora { get; set; }
    public string Categoria { get; set; }
    public string LivroId { get; set; }
    public bool Emprestado { get; set; }

    public  ICollection<Avaliacao> Avaliacoes { get; } = new List<Avaliacao>();
    public List<Comentario> Comentarios { get; set; }

    public Livro(string titulo, string autor, string editora, string categoria){
        Titulo = titulo;
        Autor = autor;
        Editora = editora;
        Categoria = categoria;
        LivroId = Guid.NewGuid().ToString();
    }

}

File: front/src/models/Livro.ts
export interface Avaliacao {

    id?: string;
    estrelas: number;
    usuario: string;
}

export interface Comentario {
    id?: string;
    texto: string;
    usuario: string;

}

export interface Livro {
    titulo: string;
    autor: string;
    editora: string;
    categoria: string;
    livroId?: string;
    emprestado?: boolean;

    avaliacoes?: Avaliacao[];
    comentarios?: Comentario[];
}
